go/vt/vttablet/tabletmanager/vdiff/table_differ.go

```diff
@@ -398,12 +398,18 @@ func (td *tableDiffer) restartTargetVReplicationStreams(ctx context.Context) err
 func (td *tableDiffer) streamOneShard(ctx context.Context, participant *shardStreamer, query string, lastPK *querypb.QueryResult, gtidch chan string) {
 	log.Infof("streamOneShard Start on %s using query: %s", participant.tablet.Alias.String(), query)
 	td.wgShardStreamers.Add(1)
+
 	defer func() {
 		log.Infof("streamOneShard End on %s", participant.tablet.Alias.String())
-		close(participant.result)
-		close(gtidch)
+		select {
```

Comment:
Defensive code to handle a panic seen in a prod cluster where the channel was closed prematurely. Unable to reproduce it using local tests though.

Reply:
if the context is done, we do not close the the channels?